README.md

```diff
@@ -9,43 +9,61 @@ JavaScript Fix this
+We want to get our bakery up and running, but there are problems with
+making sure that the right things are getting baked the right way.
 
-``` javascript
-function makeCake(ingredients) {
-  setTimeout(function() {
-    return `Made a cake with ${this.type} as the base and ${ingredients}.`
-  }, 1000)
-}
+If you run `index.html` in your browser and click either of the "make"
+links, it should go through all the right steps to make the right
+dessert, and update the status for each step as it goes.
 
-// should be
+Right now, all our baking functions are mixed up and can't figure out
+their `this`. We've got some work to do before we can open our bakery.
 
-function makeCake(ingredients) {
-  setTimeout(function() {
-    return `Made a cake with ${this.type} as the base and ${ingredients}.`
-  }.bind(this), 1000)
-}
+* Inside the `makePie` function, "borrow" the `decorate` function from `cake` and make it available
+  to `pie` through `pie.decorate()` so it can be executed later.
 
-// so that it can be called like
+* For the `bake`, `mix`, and `cool` functions, make
+sure that the function for the next step is being called with the
+correct context, and that the proper `updateFunction` is being called to
+update the status. You'll need to use `call` inside these functions to
+get the tests to pass. **Hint:** Remember that the `function` argument to `setTimeout` *also*
+needs to be bound to the proper context!
 
-makeCake.call({ type: 'chocolate' }, "strawberries, cherries, and cinnamon")
-```
+* Write your own code for the `makeDessert` function
+that will decide based on which link was clicked whether to `makePie` or
+`makeCake`. **Hint:** You shouldn't need to alter the code in the
+`document.addEventListener` block, but remember that events also set
+`this` when they are triggered from the DOM.
 
-This example obviously has less punch now that arrow functions are available. (Perhaps mention in the intro that students can avoid a lot of `bind()` by using arrow functions?)
+* Create a version of `updateStatus` inside of `makePie` and `makeCake` with a correct `this` context, representing either the `pie` or `cake` `<div>` respectively, that you can pass around to the other functions so that each one can execute it and ensure that the right DOM elements are getting updated at each step. You shouldn't need to change `updateStatus` at all.
 
-Another potential test manipulates things in the DOM, but messes up the `this` in the callback
+* There's also some bugs in the code, so make sure the tests are passing and run the page to see it in action and make sure it works! Some tests already pass. Part of your job is to make sure they still pass at the end!
+
+Each step of the way, you should be using what you know about `call`,
+`bind`, and `apply` to make each function set the context for the next
+function.
+
+**Advanced:** If you get tired of manually binding context to every
```

Comment:
Students will have learned about arrow functions earlier in the curriculum (this is at the end of Unit 2: https://github.com/learn-co-curriculum/javascript-arrow-functions), so we can probably just mention this as a reminder :)